src/careamics/lightning/callbacks/prediction_writer_callback/write_strategy.py

```diff
@@ -0,0 +1,394 @@
+
+    @property
+    def last_tiles(self) -> list[bool]:
+        """
+        List of bool to determine whether each tile in the cache is the last tile.
+
+        Returns
+        -------
+        list of bool
+            Whether each tile in the tile cache is the last tile.
+        """
+        return [tile_info.last_tile for tile_info in self.tile_info_cache]
+
+    def write_batch(
+        self,
+        trainer: Trainer,
+        pl_module: LightningModule,
+        prediction: tuple[NDArray, list[TileInformation]],
+        batch_indices: Optional[Sequence[int]],
+        batch: tuple[NDArray, list[TileInformation]],
+        batch_idx: int,
+        dataloader_idx: int,
+        dirpath: Path,
+    ) -> None:
+        """
+        Cache tiles until the last tile is predicted; save the stitched prediction.
+
+        Parameters
+        ----------
+        trainer : Trainer
+            PyTorch Lightning Trainer.
+        pl_module : LightningModule
+            PyTorch Lightning LightningModule.
+        prediction : Any
+            Predictions on `batch`.
+        batch_indices : sequence of int
+            Indices identifying the samples in the batch.
+        batch : Any
+            Input batch.
+        batch_idx : int
+            Batch index.
+        dataloader_idx : int
+            Dataloader index.
+        dirpath : Path
+            Path to directory to save predictions to.
+        """
+        dls: Union[DataLoader, list[DataLoader]] = trainer.predict_dataloaders
+        dl: DataLoader = dls[dataloader_idx] if isinstance(dls, list) else dls
+        ds: IterableTiledPredDataset = dl.dataset
+        if not isinstance(ds, IterableTiledPredDataset):
+            raise TypeError("Prediction dataset is not `IterableTiledPredDataset`.")
+
+        # cache tiles (batches are split into single samples)
+        self.tile_cache.extend(np.split(prediction[0], prediction[0].shape[0]))
+        self.tile_info_cache.extend(prediction[1])
+
+        # save stitched prediction
+        if self._have_last_tile():
+
+            # get image tiles and remove them from the cache
+            tiles, tile_infos = self._get_image_tiles()
+            self._clear_cache()
+
+            # stitch prediction
+            prediction_image = stitch_prediction_single(
+                tiles=tiles, tile_infos=tile_infos
+            )
+
+            # write prediction
+            sample_id = tile_infos[0].sample_id  # need this to select correct file name
+            input_file_path = get_sample_file_path(ds=ds, sample_id=sample_id)
+            file_path = create_write_file_path(
+                dirpath=dirpath,
+                file_path=input_file_path,
+                write_extension=self.write_extension,
+            )
+            self.write_func(
+                file_path=file_path, img=prediction_image[0], **self.write_func_kwargs
+            )
+
+    def _have_last_tile(self) -> bool:
```

Comment:
My (non-native) brain would name the method `_has_last_tile`. Not super important, maybe just a matter of taste!

Reply:
Yeah "has last tile" probably makes more sense I will update.